### ISSUE TITLE
Breaking changes in pyasn1 >= 0.4.1

### ISSUE THREAD

**Issue by liutec** [cannatag/ldap3#449]
Hello, 

Since version 0.4.1, legacy `None` initializer support has been removed in pyasn1 https://github.com/etingof/pyasn1/pull/91

This introduces the following issue:

```
[...]
search_filter = '(&(objectCategory=person)(!(userAccountControl:1.2.840.113556.1.4.803:=2))(distinguishedName=' + dn + '))'
if conn.search(search_base=host_dn, search_filter=search_filter, search_scope=ldap3.SUBTREE, attributes=fields.keys()):
[...]
```

```userAccountControl:1.2.840.113556.1.4.803:=2``` causes `dn_attributes` to retain the [default value of `None`](https://github.com/cannatag/ldap3/blob/master/ldap3/operation/search.py#L111) for the matching filter [initialization](https://github.com/cannatag/ldap3/blob/master/ldap3/operation/search.py#L267)

```
[...]pyasn1\type\univ.py", line 254, in prettyIn
    return int(value)
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```

Either the requirements should be changed to ```pyasn1>=0.1.8,<0.4.1``` or at least the default value for `dn_attributes` should be changed to `False`.

Sorry I did not create a PR but I am not familiar with the library code.

**Comment by gregn610** [comment]
I'm seeing this too, even with pyasn1==0.4.2. I think it's being triggered by a paged_search() with no results.

    > /usr/local/lib/python3.6/site-packages/pyasn1/type/univ.py(254)prettyIn()
    -> return int(value)
    (Pdb) w
      /usr/local/lib/python3.6/unittest/case.py(59)testPartExecutor()
    -> yield
      /usr/local/lib/python3.6/unittest/case.py(601)run()
    -> testMethod()
      /src/tests/tests_engine.py(163)test_sync()
    -> engine.sync()
      /src/padnag/engine.py(81)sync()
    -> self.role_transforms
      /src/padnag/directoryservice.py(119)get_targets_state()
    -> ['objectClass', 'objectGUID', role_attribute])
      /src/padnag/directoryservice.py(80)search_ad()
    -> generator=False,
      /usr/local/lib/python3.6/site-packages/ldap3/extend/__init__.py(125)paged_search()
    -> paged_criticality)
      /usr/local/lib/python3.6/site-packages/ldap3/extend/standard/PagedSearch.py(117)paged_search_accumulator()
    -> paged_criticality):
      /usr/local/lib/python3.6/site-packages/ldap3/extend/standard/PagedSearch.py(63)paged_search_generator()
    -> None if cookie is True else cookie)
      /usr/local/lib/python3.6/site-packages/ldap3/core/connection.py(762)search()
    -> self.server.schema if self.server else None)
      /usr/local/lib/python3.6/site-packages/ldap3/operation/search.py(370)search_operation()
    -> request['filter'] = compile_filter(parse_filter(search_filter, schema, auto_escape, auto_encode).elements[0])  # parse the searchFilter string and compile it starting from the root node
      /usr/local/lib/python3.6/site-packages/ldap3/operation/search.py(230)compile_filter()
    -> boolean_filter[pos] = compile_filter(element)
      /usr/local/lib/python3.6/site-packages/ldap3/operation/search.py(267)compile_filter()
    -> matching_filter['dnAttributes'] = DnAttributes(filter_node.assertion['dnAttributes'])
      /usr/local/lib/python3.6/site-packages/pyasn1/type/univ.py(103)__init__()
    -> base.AbstractSimpleAsn1Item.__init__(self, value, **kwargs)
      /usr/local/lib/python3.6/site-packages/pyasn1/type/base.py(240)__init__()
    -> value = self.prettyIn(value)
    > /usr/local/lib/python3.6/site-packages/pyasn1/type/univ.py(254)prettyIn()
    -> return int(value)


hth
/g

**Comment by dirkjanm** [comment]
I'm having the same issue with the following query:
`(&(objectCategory=person)(objectClass=user)(sAMAccountName=TEST)(memberOf:1.2.840.113556.1.4.1941:=CN=Domain Admins,CN=Users,DC=TEST,DC=local))`

**Comment by cannatag** [comment]
Please check code in dev. It should work now.

**Comment by NeilHanlon** [comment]
I can confirm #458 fixes the issue (manually patched a system running apache-airflow that uses ldap for auth.

**Comment by cannatag** [comment]
Thanks, the fix will be released in the upcoming version of ldap3.

**Comment by mcosta74** [comment]
Hi Giovanni,

Any ETA for the next version?

**Comment by cannatag** [comment]
It should be next week. I’m waiting for  confirmations of other fIxes.

**Comment by mcosta74** [comment]
Thanks a lot for the info and for your great job.

BTW, Italians do it better 😉

Inviato dal mio telefono Windows 10

Da: Giovanni Cannata
Inviato: giovedì 4 gennaio 2018 21:30
A: cannatag/ldap3
Cc: Massimo Costa; Comment
Oggetto: Re: [cannatag/ldap3] Breaking changes in pyasn1 >= 0.4.1 (#449)

It should be next week. I’m waiting for confirmations of other fIxes.

**Comment by cannatag** [comment]
Sorry, I'm still waiting for responses for other fixes. Anyway I will release the 2.4.1 in the next weekend.

**Comment by cannatag** [comment]
Released in version 2.4.1
